src/node/channels.h

```diff
@@ -66,7 +91,12 @@ namespace ccf
       {
         // Set local recv nonce to received nonce only if verification is
         // successful
-        local_recv_nonce.store(recv_nonce);
+        while (local_nonce < recv_nonce->nonce)
```

Comment:
As discussed, this can go for now as the receiving thread is always the same for now.